src/main/java/ru/l3r8y/prefixed/PrefixedCop.java

```diff
@@ -23,21 +23,130 @@
+import org.apache.maven.project.MavenProject;
+import ru.l3r8y.prefixed.annotation.Prefixed;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
 
 /**
  * Prefixed cop.
+ * This class designed with focus on performance, this is
+ * why code may look ugly to you.
+ *
  * @since 0.0.0
  */
 @Mojo(name = "prefixed", defaultPhase = LifecyclePhase.VERIFY)
 public final class PrefixedCop extends AbstractMojo {
 
+    /**
+     * Thread pool to scan project.
+     */
+    private static final ExecutorService EXECUTORS = Executors.newVirtualThreadPerTaskExecutor();
+
+    /**
+     * The project.
+     */
+    @Parameter(defaultValue = "${project}")
+    private MavenProject project;
+
+    /**
+     * The base package.
+     */
+    @Parameter(property = "basePackage", defaultValue = "${project.groupId}")
+    private String basePackage;
+
+    /**
+     * Fail on error flag.
+     */
+    @SuppressWarnings("PMD.ImmutableField")
+    @Parameter(defaultValue = "true")
+    private boolean failOnError = true;
+
     @Override
-    public void execute() throws MojoExecutionException, MojoFailureException {
-        this.getLog().info("Boom!");
+    public void execute() throws MojoExecutionException {
+        final String outputDir = this.project.getBuild().getOutputDirectory();
+        final File output = new File(outputDir);
+        if (!output.exists()) {
+            this.getLog().info(String.format("Output directory does not exist: '%s'", outputDir));
+            return;
+        }
+        this.getLog().info(String.format("Scanning compiled classes in: '%s'", outputDir));
+        final ClassGraph graph = new ClassGraph()
+            .overrideClasspath(outputDir)
+            .acceptPackages(this.basePackage)
+            .enableClassInfo()
+            .enableAnnotationInfo();
+        try (final ScanResult scanResult = graph.scan(PrefixedCop.EXECUTORS, Runtime.getRuntime().availableProcessors())) {
+            final List<ClassInfo> prefixedInterfaces = scanResult
+                .getClassesWithAnnotation(Prefixed.class)
+                .stream()
+                .filter(ClassInfo::isInterface)
+                .toList();
+            this.getLog().info("Total interfaces found: " + prefixedInterfaces.size());
+            final List<String> errors = new ArrayList<>(0);
+            for (final ClassInfo iface : prefixedInterfaces) {
+                final AnnotationInfo prefixedAnno = iface.getAnnotationInfo().get(Prefixed.class.getName());
+                final AnnotationParameterValue prefixParam = prefixedAnno.getParameterValues().get("prefix");
+                if (prefixParam == null) {
+                    this.handleMissingPrefix(iface.getName());
+                    continue;
+                }
+                final String expectedPrefix = prefixParam.getValue().toString();
```

Comment:
@l3r8yJ Let's wrap this logic into `Expected`, `Prefix`, and make a rival between them

Reply:
@l3r8yJ The idea is simple: let's create three objects: `Expected` - encapsulates expected `prefix`, `Prefix` - encapsulates an actual prefix. Those objects form third object, that compares them.

Reply:
@h1alexbel don't you think that maybe it's overcomplicated for simple string value?

Reply:
@l3r8yJ maybe we can pass Prefixed annotation as `Prefix` then?

Reply:
@h1alexbel do you suggesting rename `RequirePrefix` into `Prefix`?